### ISSUE TITLE
Naming scheme - autogeneration files

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#941]
Concerning the naming scheme:
I prefer the `.in.extension` version, for the following reasons:
* This way it is quite easy to see what files are auto-generated - if a `bla.in.[a-z]*` is available, `bla` is auto-generated.
* The generated file should not be prefixed with generated_ (as I used to do in the past) as I don't think a user should have to care. A developer should look for the `.in.ext` postfix to check if there is a source file
* If finding the file with tab-completion, it is much easier to find the actual source file, and might help the developer remember that the file is auto-generated.
* The name rule is much simpler - no replace`.` with `_` needed - just add or remove `.in.ext`
* Unlike `.in` as I used in the cython branch, the language is easier to see with adding the `.ext`
* I think more files might be generated in the future - thus prefixing with `generated_` would not necessary be nice

Other schemes in use:
`gen_fieldops.py` -> `generated_fieldops.cxx`
`gen_fieldops.jinja` -> `generated_fieldops.cxx`
`setup.py.in` -> `setup.py`

All arguments are welcome :)

**Comment by d7919** [comment]
I agree that having a consistent approach is important, but I'm not sure what the "best" choice is.

**Comment by ZedThree** [comment]
Generally, `<filename>.in` files are pure data, input for some kind of driver program/script that then spits out `<filename>`.

Having `<filename>.in.py` be the driver as well as the input data is unusual. I'd like to separate the input files from the driver files, and ideally one would only need to modify the input files and run the driver in order to change the output files.

Dropping the `generated_`/`gen_` prefix makes sense though. I'm happy for the Jinja template files to be named either `<filename>.in` or `<filename>.jinja`, I don't have a preference there, but we should stick to one.

Ideally, what I'd like to see is that we only have a single driver for all the different generated code, but that is probably quite tricky. We should look at how other projects do it -- Chromium use Jinja templates to generate C++ files, and they have a reasonably nice style guide for it, so it would make sense to learn from what they do.

**Comment by dschwoerer** [comment]
I am not sure what you mean with *pure data*.
Is jinja *pure data*? It seems to be Turing complete ...

I guess on whether you need to change the templates or the driver is a question of what you are going to change? If you want to add new fields to the fieldops - would you want to do that purely in jinja?

Is there a way to look at the chromium source code in an acceptable way? The official documentation suggests to run untrusted code on my computer to get the source code.
